tests/test_freeimage.py

```diff
@@ -3,35 +3,65 @@
 
 import os
 import sys
+import shutil
 
 import numpy as np
 
 from pytest import raises, skip
 import pytest
-from imageio.testing import run_tests_if_main, get_test_dir, need_internet
 
 import imageio
 from imageio import core
-from imageio.core import get_remote_file, IS_PYPY
+from imageio.core import IS_PYPY
 
 
-test_dir = get_test_dir()
+@pytest.fixture(scope="module")
+def setup_library(tmp_path_factory, image_cache):
 
+    # Checks if freeimage is installed by the system
+    from imageio.plugins.freeimage import fi
+
+    use_imageio_binary = False
+    if not fi.has_lib():
+        use_imageio_binary = True
 
-def setup_module():
     # During this test, pretend that FI is the default format
     imageio.formats.sort("-FI")
 
-    # This tests requires our version of the FI lib
-    try:
-        imageio.plugins.freeimage.download()
-    except core.InternetNotAllowedError:
-        # We cannot download; skip all freeimage tests
-        need_internet()
-
-
-def teardown_module():
-    # Set back to normal
+    if use_imageio_binary:
+
+        # Setup from image_cache/freeimage
+        ud = tmp_path_factory.getbasetemp() / "userdir"
```

Comment:
```suggestion
        ud = tmp_path_factory.mktemp("userdir")
```

this way pytest will take care of the cleanup.

Reply:
Excellent.